### DISCUSSION TITLE
GCS access By WorkLoad Identity

### DISCUSSION THREAD

**Discussion by vipul25-ds** [milvus-io/milvus#35532] [Q&A and General discussion]
Hi All,

I have added below entries in my Yaml to enable GCS storage for my Milvus instance. I could see the insert_log and status_log, but not able to see the collection records/ rows  getting stored in GCS.

1. 
cluster:
    enabled: true
service:
    type: LoadBalancer
minio:
    enabled: false
serviceAccount:
    create: false
    name: milvus-gcs-access-sa
externalS3:
    enabled: true
    host: storage.googleapis.com
    port: 443
    rootPath: milvus/my-release
    bucketName: milvus-testing-nonprod
    cloudProvider: gcp
    useSSL: true
    useIAM: true
2. Tried collection.flush as well 

Two Questions:
1. May I know what else would be needed to make this work?
2. I have a scenario, where I want to store only older set of records on GCS bucket, where as the newer records should get inserted in the memory. may I know how would I achieve that

**Comment by xiaofan-luan** [comment]
1. by trigger flush you should see stats log and insert log in GCS
2. Milvus already handled that, newer data is stored in Kafka and older data is stored in GCS after flush.
Can you explain a little bit more about your use case?

**Reply by vipul25-ds** [reply]
@xiaofan-luan 
**1. by trigger flush you should see stats log and insert log in GCS.** 
    My response; I am able to see the insert_log and stats log, but Iam not able to see the records/rows which I am saving in collection

**Milvus already handled that, newer data is stored in Kafka and older data is stored in GCS after flush.
Can you explain a little bit more about your use case?**

My Answer; I want to create a warm and cold storage, based on the records age. for eg; if the record is 60 days old it should be stored in GCS and all the newer records should be inserted in Memory

**Reply by xiaofan-luan** [reply]
> but Iam not able to see the records/rows which I am saving in collection

1. "but Iam not able to see the records/rows which I am saving in collection", how did you verify that? Did you do count or did you search/query and didn't find result.

2. this won't be supported for now. because we search happened, we don't know if the record is in cold or hot. On Zilliz cloud we support LRU and tiered storage but you will still expect to see long latency (500ms) when searching on cold data

**Reply by vipul25-ds** [reply]
@xiaofan-luan 
**"how did you verify that?":**
I am checking inside the same 'rootPath'. I can see two folders "Insert_log" and "status_log", but not able to see the records there. 
**"Did you do count or did you search/query and didn't find result "**
I do see the result when I am doing count/ search results, but the records are  not visible in the GCS bucket location.

**this won't be supported for now. because we search happened, we don't know if the record is in cold or hot. On Zilliz cloud we support LRU and tiered storage but you will still expect to see long latency (500ms) when searching on cold data**
Lets suppose there are two collections A and B. is there a way I can store all the records in collection A in Memory and all the records in Collection in B in GCS bucket/

**Reply by xiaofan-luan** [reply]
> @xiaofan-luan **"how did you verify that?":** I am checking inside the same 'rootPath'. I can see two folders "Insert_log" and "status_log", but not able to see the records there. **"Did you do count or did you search/query and didn't find result "** I do see the result when I am doing count/ search results, but the records are not visible in the GCS bucket location.
> 
> **this won't be supported for now. because we search happened, we don't know if the record is in cold or hot. On Zilliz cloud we support LRU and tiered storage but you will still expect to see long latency (500ms) when searching on cold data** Lets suppose there are two collections A and B. is there a way I can store all the records in collection A in Memory and all the records in Collection in B in GCS bucket/

all data will be stored in GCS bucket. but you can select to put data cached on local ssd or in memory or not loaded it for search

**Comment by xiaofan-luan** [comment]
you can offer detailed logs (especially datacoord or datanode so we can see if there are some hints)

**Reply by xiaofan-luan** [reply]
trigger flush manually and you should see some data in insert_log stats_log
